### PR TITLE
Combination and Graph algorithms

### PR DESCRIPTION
**Pull request by adi1307**
Three algorithms have been added which are as follows:-
a) Combination algorithm
b) Detecting cycle in directed graph
c) Detecting cycle in undirected graph

### CODE REVIEW

File: Algorithms/Combination/combination_by_eluer_method.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ 	b /= 2;
+ 	}
+ return x;
+}
+/* Modular Multiplicative Inverse
+ Using Euler's Theorem
+ a^(phi(m)) = 1 (mod m)
+ a^(-1) = a^(m-2) (mod m) */
+ll InverseEuler(ll n, ll MOD)
+{
```
Comment (Bhavnaharitsa) [inline Algorithms/Combination/combination_by_eluer_method.cpp:42]: avoid single variable declaration

File: Algorithms/Combination/combination_by_eluer_method.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ return (f[n]*((InverseEuler(f[r], MOD) * InverseEuler(f[n-r], MOD)) % MOD)) % MOD;
+}
+
+int main(){
+	f[0] = 1;
+	for(int i = 1 ; i <= 100000 ; i++)
+		f[i] = (f[i-1]*i)%INF;
+    ll n,r,ans;
+    cout<<"Enter the value of N\n";
+    cin>>n;
```
Comment (Bhavnaharitsa) [inline Algorithms/Combination/combination_by_eluer_method.cpp:58]: here too

File: Algorithms/Graphs/cycle_in_directed_graph.cpp [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+int main()
+{
+    // Create a graph given in the above diagram
+    int n,m,a,b,i;
+    cout<<"Enter the number of vertices\n";
+    cin>>n;
+    cout<<"Enter the number of edges\n";
+    cin>>m;
```
Comment (Bhavnaharitsa) [inline Algorithms/Graphs/cycle_in_directed_graph.cpp:105]: use void main()

File: Algorithms/Graphs/cycle_in_directed_graph.cpp [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        if (isCyclicUtil(i, visited, recStack))
+            return true;
+
+    return false;
+}
+
+int main()
+{
+    // Create a graph given in the above diagram
+    int n,m,a,b,i;
```
Comment (Bhavnaharitsa) [inline Algorithms/Graphs/cycle_in_directed_graph.cpp:101]: no single variable declaration

File: Algorithms/Graphs/cycle_in_undirected_graph.cpp [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+#include<iostream>
+#include <list>
+#include <limits.h>
+using namespace std;
+
+// Class for an undirected graph
+class Graph
+{
+    int V;    // No. of vertices
```
Comment (Bhavnaharitsa) [inline Algorithms/Graphs/cycle_in_undirected_graph.cpp:33]: change variable name to no_of_vertices

**Review by chanmol1999 — APPROVED** [review]

**Pull request merged**